### ISSUE TITLE
VLBI-Investigating-the-effect-of-averaging-vs-correlator-windowing-functions

### ISSUE THREAD

**Issue by atemkeng** [atemkeng/Windowing-Functions#15]
This shows us the effect of averaging across 1.4Khz and 3s. I was wondering if it is not a good idea to meet some one who specializes in VLBI so that we could really know the idea of the re-sampling channel width and time. I wonder how on Iniyan's dataset they average across 16MHz and 4s.
![int3smhz](https://cloud.githubusercontent.com/assets/6806881/9060071/9dc537a2-3ab0-11e5-9da7-fcbb91ed72a3.png)

**Comment by atemkeng** [comment]
I am just playing with the hires integration time and channel width, also the lores re-sampling time and channels width. 
![int3s30mhz](https://cloud.githubusercontent.com/assets/6806881/9062564/825d0be4-3ac3-11e5-8409-80237d2f4f3f.png)

**Comment by atemkeng** [comment]
The following shows a 1s and 10KHz compression for sinc and Avg
![vlbi_1s_10khz](https://cloud.githubusercontent.com/assets/6806881/9141071/247adc66-3d36-11e5-8090-5c4c447c74a1.png)

This plot  shows a 5s compression for sinc and 1s compresion for averaging both with the same channels compresion rate, 10KHz.
![sinc-1x1-5s](https://cloud.githubusercontent.com/assets/6806881/9141150/eb485c56-3d36-11e5-8fe2-f09e201f7d67.png)

From what now I understand, we need to compress more with BDWF in the VLBI regime and therefore we reduce noise and recover more FoV. So, I have some scripts still running 

@o-smirnov

**Comment by o-smirnov** [comment]
What's the frequency, 1.4 GHz?

**Comment by atemkeng** [comment]
The frequency is 4.9 GHz as in Inyan dataset. You will surely have something better than this at 1.4GHz

**Comment by o-smirnov** [comment]
Hmmm, I'm not sure about 10kHz/1s then. Remember that we get best results
when the degree of bandwidth and time averaging is comparable. At 1.4 GHz,
1 MHz ~ 10s. At 5 GHz, it would be (5/1.4)*1 MHz ~ 10s i.e. roughly 3 MHz ~
10s. So maybe 30 kHz ~ .1s would be a better combination? But do
double-check my numbers, as it's late on a Friday and I may be barking up
the wrong tree.

**Comment by atemkeng** [comment]
ok will do that, I am still runing some few combinations. But I evaluted everything. averaging can only recover a FoV of radius approximately 28Arcsec with 1s and 10KHz integrations at this frequency. So, now I have to find a window and compresions factor than can do better than that.

**Comment by atemkeng** [comment]
From my investigations, these are the best regimes, (3s, 1MHz), (5s, 1.8MHz) and so on. The VLBI calculator confirm  averaging smearing rate on the following plots at 4.9GHz with the longest baseline been 9000Km. The noise getting up is trivial in this VLBI. In fact, here all our windows have high time and frequency resolution, so here there is nothing like boxcar on the shorter baselines as it was with an array like JVLA or an array with long and short baselines. Notice that this VLBI shortest baseline is ~78 time longer  than the JVLA C longest baseline.
BDWF will be better  for VLBI that include also short baselines, in term of noise ratio. @o-smirnov 

![vlbi_3s_1mhz](https://cloud.githubusercontent.com/assets/6806881/9174781/51d70d6c-3f82-11e5-9dcb-ef0622cca55f.png)
![vlbi_5s_1 8mhz](https://cloud.githubusercontent.com/assets/6806881/9174783/544a8268-3f82-11e5-90e6-f3064abc10de.png)

**Comment by o-smirnov** [comment]
The sheer flatness of the sinc curve is mighty impressive! Still, 2.88 is a hefty noise penalty, so if we can trade off some FoV loss against lower noise I'd do it. What's the FoV parameter of the sinc set to in this case? What happens if you make the FoV smaller? 

Also, what about lower averaging values? 1s 300kHz and .1s 30 kHz would be interesting to see.

**Comment by atemkeng** [comment]
![vlbi_1s_358khz](https://cloud.githubusercontent.com/assets/6806881/9220533/95729c50-40e3-11e5-8fa2-b3110548b073.png)

**Comment by atemkeng** [comment]
Things are showing up now clearly. Need to simulate now many and observe the behaviour.  Now we can control the FoV on VLBI, need to confirm this with many simulation. The plot shows a FoV of 1deg across. 
![vlbi_0 5s_50khz](https://cloud.githubusercontent.com/assets/6806881/9255661/16643ff0-41eb-11e5-8bcb-c50ddb9b5789.png)

@o-smirnov

**Comment by atemkeng** [comment]
Hires 0.01s integration and 2KHz bandwidth

![int1s-bw116khz](https://cloud.githubusercontent.com/assets/6806881/9577994/a47f5992-4fe6-11e5-899c-618c8430a0fe.png)
![int1s-bw116khz](https://cloud.githubusercontent.com/assets/6806881/9577998/aade7ebc-4fe6-11e5-972a-5f3c8709254b.png)
![figure_1](https://cloud.githubusercontent.com/assets/6806881/9578073/5655743a-4fe7-11e5-87ab-c319559956db.png)

@o-smirnov

**Comment by o-smirnov** [comment]
Could you please point me to the VLBI MS you're using? I'm confused, this 

https://github.com/atemkeng/Windowing-Functions/blob/master/FinalData/Data-VLBI/ANTENNAS-conf.txt

...says the shorttest baseline is 267m, but how can this be? None of these stations are that close to each other. Don't you mean 267km? 

It may be useful to make a uv-coverage plot for the VLBI experiment.

**Comment by atemkeng** [comment]
Hi Prof, Sorry for the late reply, this is because Yesterday after the
workshop there were no internet, I am just seeing my emails now. Let me
point you on the VLBI MS, but let me check again. Thank you

**Comment by atemkeng** [comment]
Thank you very much, it is very nice the paper, I had read and made all the corrections in red and the references. There are two references that I could not find, I mentioned in red in the paper. The VLBI shortest baseline is 266.519Km I made an error when I was writing the readMe. I am doing some more simulations for the VLBI, not just a snapshot. So, I will prefer to use this MS for the coverage elwood:/home/atemkeng/VLBI_5s_16MHZBIS/DATA_RESULT/vlbi-hires-0.01s-500Hz.MS/

**Comment by atemkeng** [comment]
A re-sampling over 1s and 12.5KHz with sinc and bessel. The Hires is sampled at 0.01s and 500Hz channels widths.

![int1s-bw12khz-overlap](https://cloud.githubusercontent.com/assets/6806881/9850744/f2e38db0-5af4-11e5-8aa3-833179cceeba.png)

**Comment by atemkeng** [comment]
This combinations improve up to 115KHz@o-smirnov 
![int0 5s-115khz-sinc-bessel](https://cloud.githubusercontent.com/assets/6806881/9941804/ee726fb4-5d77-11e5-8453-0994c4a05862.png)

**Comment by o-smirnov** [comment]
Now that's a killer result! Be sure to show this to the JIVE guys. This is a snapshot I presume though?

Why this time/freq combination? Wouldn't things work better around 0.5s/50kHz, or 1s/100kHz, ensuring "square" bins (as per the arguments in the paper). This is 1.6GHz observing freq, right?

Also, need to get a few more points close to 0, because otherwise you cannot tell at all what the averaging FoV is -- first data point is already at 0.6.

**Comment by atemkeng** [comment]
Yeah the results are very promising! I tried 1s and 115KHz but the noise penalty was slightly above 2 what I do not want to see. Yes I am very sure,  you can try it also. I will try 1s and 50Khz but as my discussion with Zolt, to achieve a wide FoV (~30arcmin) with VLBI at 1.6GHz you need 0.01s  resampled in time and 11KHz in frequency. So for 0.1s resampled in time it is already a compression rate of  factor 10 in time and 115KHz it is approximately a factor of 10 in frequency. My aims were to achieve at least a compression rate factor of 10 in  frequency. I think this is not a snapshot, 5.5s synthesis with this VLBI (with very long baselines) all the baselines will sweep a complete ellipse during this time.

**Comment by o-smirnov** [comment]
BTW, is that a frequency overlap of "1.4"? Really? Why such a strange value?

**Comment by atemkeng** [comment]
No reason, I just have that because I am playing around to find better
options.

**Comment by o-smirnov** [comment]
4x2 will cause less questions from the referee. And BTW, I now realize that with overlap taken into account, you do have more square-ish bins. 0.5sx4 and 115x1.4 is closer to square... but 4x2 will be even closer.

**Comment by o-smirnov** [comment]
In the data files currently checked in, all the noise penalties appear to be 1 or 1.01. This is impossible.

**Comment by atemkeng** [comment]
These are the best cases for VLBI. The data for all these plots are saved on github, VLBI repertory.

Hires of 0.001s integration time and  4.75s synthesis at 16GHz with a bandwidth of 150000Hz channelized into 300 channels each of width 500Hz. 

1-resampled 0.01s in time and 12.5KHz in frequency

![figure0 01s_12 5khz](https://cloud.githubusercontent.com/assets/6806881/10045658/d03059f8-6204-11e5-83f2-4598add95f5c.png)

2-resampled 0.1s in time and 25KHz in frequency
![figure0 1s_25khz](https://cloud.githubusercontent.com/assets/6806881/10045662/db669832-6204-11e5-8b8a-86cce68d2d8e.png)

3-resampled 0.25s in time and 25KHz in frequency
![figure0 25s_25khz](https://cloud.githubusercontent.com/assets/6806881/10045664/dfce2a52-6204-11e5-9927-8b42a2a6ec0c.png)

**Comment by o-smirnov** [comment]
So is this plot: https://github.com/atemkeng/Windowing-Functions/issues/15#issuecomment-141169800 still correct? Why is it so "flat" along the top, and the most recent ones aren't?

**Comment by atemkeng** [comment]
Yes if you run with the same conditions you will have the plots #15, but that noise ratio is not correct. You can have flat as you want, the problem is the noise penalty.

**Comment by atemkeng** [comment]
"Assume 1.6 GHz, FoV tuned to 5' radius. The question is, what is the
_largest_ integration time and channel width we can use and have a
noise penalty of no more than 1.5 (for the best-performing filter,
i.e. bessel-4x3). "

All will depend if you want high time resolution or frequency. The two plots answer your questions (see title of the plots). 

Resampling time = 0.5s, Resampling frequency 13KHz

![figure_05s](https://cloud.githubusercontent.com/assets/6806881/10729688/d502149a-7bf3-11e5-846c-1a77485fff40.png)

Resampling time =0.25s and Resampling frequency 23KHz

![figure_025s](https://cloud.githubusercontent.com/assets/6806881/10729561/fba14d42-7bf2-11e5-81bc-e53ef4860282.png)

@o-smirnov

**Comment by o-smirnov** [comment]
OK, and if we allow a noise penalty of up to 2?

**Comment by atemkeng** [comment]
Resampling time = 0.5s, Resampling frequency 26KHz

![figure_05s_26khz](https://cloud.githubusercontent.com/assets/6806881/10878050/d21f5796-8153-11e5-9f12-e89ef757a4b2.png)

Resampling time = 0.25s, Resampling frequency 52KHz

![figure_25s_52khz](https://cloud.githubusercontent.com/assets/6806881/10878069/f06dd632-8153-11e5-9207-3b71f1403efc.png)

@o-smirnov

**Comment by atemkeng** [comment]
integration time = 10ms, Resampling frequency 0.5MHz

![figure_10ms_05mhz](https://cloud.githubusercontent.com/assets/6806881/11038283/d8ac6e84-870a-11e5-8c7f-04a99b36d1d3.png)

@o-smirnov

**Comment by atemkeng** [comment]
resample time 0.01s, resample channels widths = 0.125MHZ

![figure_10ms_012mhz](https://cloud.githubusercontent.com/assets/6806881/11165577/d8ef7a48-8b1b-11e5-988e-66453c742546.png)

resample time 0.01s , resample channels widths = 0.25MHZ 
![figure_10ms_025mhz](https://cloud.githubusercontent.com/assets/6806881/11165584/1289369a-8b1c-11e5-95c1-8c604ca86442.png)

resample time 0.25s  , resample channels widths = 0.05MHZ=50kHz
![figure_025s_50khz](https://cloud.githubusercontent.com/assets/6806881/11165587/331aa916-8b1c-11e5-8167-5fc90ceb5b44.png)

resample time 0.5s , resample channels widths = 0.025MHZ=25kHz 
![figure_05s_25khz](https://cloud.githubusercontent.com/assets/6806881/11165590/44de3654-8b1c-11e5-9172-8357923b705d.png)

Find the data here and the readME.
https://github.com/atemkeng/Windowing-Functions/tree/master/FinalData/Data-VLBI/Data-1.6GHz-VLBI-0.01s-bandwidth-0.12MHz-0.25MHz

https://github.com/atemkeng/Windowing-Functions/tree/master/FinalData/Data-VLBI/Data-1.6GHz-VLBI-0.25s-0.5s-bandwidth-0.05MHz-0.025MHZ

@o-smirnov

**Comment by o-smirnov** [comment]
Those are some exciting R values! Full synthesis I assume? But we don't have airy2 described in Section 3, so we can't just drop it into the paper at that point unannounced -- either add a subsection on it to Sect 3, or redo with bessel.

Question, is airy2 then significantly better than bessel in this case? Can you put avg, airy2 and bessel on the same plot? (If I'm the referee, I'll ask why the rest of the paper is not using airy2...)

Finally, please do a 5x1 and 10x1 case for 10ms, 0.125 and 0.25MHz. If they need spectral resolution for dedispersion, then overlap in frequency becomes undesirable. So we need to see how well it works without frequency overlap. On the other hand, I want to see what happens when you increase overlap in time.

**Comment by atemkeng** [comment]
Sorry for that confusion, airy2="bessel2"

I rely do not know if these parameters are snapshot for VLBI or full synthesis. 

From your new sets of questions, I changed again the hires number of timeslots and channels.

These are the settings:

Hires:
integration time = 0.001s, number of timeslots = 100
channels widths = 500KHz, number of channels = 2500

From these:
I fixed 10 timeslots to average and  overlap_left=45, overlap_rigth=45
I fixed 500 channels to average  and overlap_left=1000, overlap_rigth=1000

LOres: (10ms, 0.12MHZ):

```
resample time 0.01s (10 timeslots averaged), resample channels widths = 0.125MHZ (250 channels averaged)

bessel-4x3:   overlap_time = 2*15, overlap_freq = 2*250

bessel-4x5:   overlap_time = 2*15, overlap_freq = 2*500

bessel-5x1:   overlap_time = 2*20, overlap_freq = 0

bessel-10x1:  overlap_time = 2*45, overlap_freq = 0 
```

![figure_10ms_012mhz](https://cloud.githubusercontent.com/assets/6806881/11168299/a2b95f58-8b92-11e5-8599-742760c0887b.png)

LOres: (10ms, 0.25MHZ):

```
resample time 0.01s (10 timeslots averaged), resample channels widths = 0.25MHZ (500 channels averaged)

bessel-4x3:   overlap_time = 2*15, overlap_freq = 2*500

bessel-4x5:   overlap_time = 2*15, overlap_freq = 2*1000

bessel-5x1:   overlap_time = 2*20, overlap_freq = 0

bessel-10x1:  overlap_time = 2*45, overlap_freq = 0 
```

![figure_10ms_025mhz](https://cloud.githubusercontent.com/assets/6806881/11168305/cc20c264-8b92-11e5-919d-3d6c87e2c624.png)

The noise penalty is correct, this is the output for my noise penalty:

{'avg-0.01s-0.12MHz': 0.0015800094,
 'avg-0.01s-0.25MHz': 0.001579888,
 'bessel-0.01s-0.12MHz-10x1': 0.0010915709,
 'bessel-0.01s-0.12MHz-4x3': 0.00211978182,
 'bessel-0.01s-0.12MHz-4x5': 0.0015492204,
 'bessel-0.01s-0.12MHz-5x1': 0.0014753153,
 'bessel-0.01s-0.25MHz-10x1': 0.0015106348,
 'bessel-0.01s-0.25MHz-4x3': 0.0028219804,
 'bessel-0.01s-0.25MHz-4x5': 0.0022566614,
 'bessel-0.01s-0.25MHz-5x1': 0.0019909651}

If you want to run it see : elwood/home/atemkeng/VLBI_5s_16MHZBIS/VLBI_overlap/TEST-0.5s-256channels/Verify

pyxis noise_estimate[noise=1.,hiresms=vlac-hires-0.001s-500kHz.MS]

this will save you a pickle file name: pickle-noise.data

@o-smirnov

**Comment by atemkeng** [comment]
To achieve the same FoV with averaging at 10% smearing, we have to re-sample 10ms in time and 15.5KHz in frequency. the compression factor can therefore be evaluate for BDWFs

![figure_10ms_15_5khz](https://cloud.githubusercontent.com/assets/6806881/11169084/71aa2baa-8bb4-11e5-8382-8f820bcbd82b.png)
